nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -0,0 +1,383 @@
+
+    protected String TRANSIENT_STORE_ERROR = "error";
+
+    protected String TRANSIENT_STORE_OUTPUT = "output";
+
+    protected String TRANSIENT_STORE_OUTPUT_BLOB = "blob";
+
+    @Context
+    protected AutomationService service;
+
+    @Context
+    protected HttpServletRequest request;
+
+    @Context
+    protected HttpServletResponse response;
+
+    @Context
+    protected CoreSession session;
+
+    @POST
+    public Object doPost(ExecutionRequest xreq) {
+        OperationResource op = (OperationResource) getTarget();
+        String opId = op.getId();
+
+        try {
+            AutomationServer srv = Framework.getService(AutomationServer.class);
+            if (!srv.accept(opId, op.isChain(), request)) {
+                return ResponseHelper.notFound();
+            }
+            final String executionId = UUID.randomUUID().toString();
+
+            // session will be set in the task thread
+            OperationContext opCtx = xreq.createContext(request, response, null);
+
+            opCtx.setCallback(new OperationCallback() {
+
+                @Override
+                public void onChainEnter(OperationType chain) {
+                    //
+                }
+
+                @Override
+                public void onChainExit() {
+                    setCompleted(executionId);
+                }
+
+                @Override
+                public void onOperationEnter(OperationContext context, OperationType type, InvokableMethod method,
+                        Map<String, Object> params) {
+                    enterMethod(executionId, method);
+                }
+
+                @Override
+                public void onOperationExit(Object output) {
+                    setOutput(executionId, (Serializable) output);
+                }
+
+                @Override
+                public OperationException onError(OperationException error) {
+                    setError(executionId, error.getMessage());
+                    return error;
+                }
+
+            });
+
+            String repoName = session.getRepositoryName();
+            NuxeoPrincipal principal = session.getPrincipal();
+
+            // TODO NXP-26303: use thread pool
+            new Thread(() -> {
+                TransactionHelper.runInTransaction(() -> {
+                    try (CloseableCoreSession session = CoreInstance.openCoreSession(repoName, principal)){
+                        opCtx.setCoreSession(session);
+                        service.run(opCtx, opId, xreq.getParams());
+                    } catch (OperationException e) {
+                        setError(executionId, e.getMessage());
+                    }
+                });
+            }).start();
+
+            String statusURL = String.format("%s/%s/status", ctx.getURL(), executionId);
+            return Response.status(HttpServletResponse.SC_ACCEPTED).location(new URI(statusURL)).build();
+
+        } catch (URISyntaxException cause) {
```

Comment:
Try to catch this very specific exception closer just around the exact location it may be thrown.